core/src/main/java/org/renjin/stats/internals/distributions/RNG.java

```diff
@@ -334,6 +633,14 @@ public double unif_rand() {
     }
   }
 
+  public static int defineSize(Vector input) {
+    int inputLength = (input.length() == 1) ? input.getElementAsInt(0) : input.length();
```

Comment:
Try to simplify this logic, it's very hard to read (and it is not appear to be correct.)